wolfcrypt/src/port/Espressif/esp32_mp.c

```diff
@@ -332,6 +332,10 @@ int esp_mp_mul(MATH_INT_T* X, MATH_INT_T* Y, MATH_INT_T* Z)
 {
     int ret = 0;
 
+#ifdef WOLFSSL_SP_INT_NEGATIVE
+    int neg = (X->sign == Y->sign) ? MP_ZPOS : MP_NEG;
```

Comment:
I recommend this just be `int neg` and keep the existing setting of it in `CONFIG_IDF_TARGET_ESP32S3`. The !CONFIG_IDF_TARGET_ESP32S3 case overrides this with a different value.

Reply:
Good point on the code inconsistency. I left the [declaration & assignment](https://github.com/gojimmypi/wolfssl/blob/2ec486ed6f46f49951092612667681d5caaba96f/wolfcrypt/src/port/Espressif/esp32_mp.c#L337) at the top and instead moved the `Z->sign` logic completely out of the IF/CONFIG sections. 

The negative flag setting is now assigned regardless of `IDF_TARGET_[n]` value at the [end of the function](https://github.com/gojimmypi/wolfssl/blob/2ec486ed6f46f49951092612667681d5caaba96f/wolfcrypt/src/port/Espressif/esp32_mp.c#L496) as updated in https://github.com/wolfSSL/wolfssl/pull/6374/commits/2ec486ed6f46f49951092612667681d5caaba96f.